### DISCUSSION TITLE
Empty wire:inital-data

### DISCUSSION THREAD

**Discussion by joval-sv** [livewire/livewire#3135] [7. Bugs (archived)]
### Description

I have a Component that is bound to a route in routes/web.php has:

```php
use App\Http\Livewire\Crm\Caso\View as Crm_ViewCaso;
...
Route::prefix('crm')->group(function(){
Route::get('/caso/{id}', Crm_ViewCaso::class);
}
```

So, when i submit a request with data, the generated HTML always has wire:initial-data as an empty string:

```html
 <div wire:id="lb9FsAqYsBqLr8lvjOiV" wire:initial-data="" class="overflow-hidden mt-44">
```

And this is where livewire.start() fails with the following error:

```
Uncaught SyntaxError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data
    Component index.js:29
    value index.js:83
    value index.js:82
    <anonymous> 1861175:1528
    EventListener.handleEvent* 1861175:1527
```

This basically points to:

```javascript
document.addEventListener("DOMContentLoaded", function () {
        window.livewire.start();
    });
```

In order to debug, I have the mount() and render() function as follows:

```php
public function mount(int $id)
    {
        $this->fill([
            'case_id' => $id,
            'ipaddr' => get_ip(), // global helper function
            'types' => array(),
            'users' => array(),
        ]);
        $this->update_caso();
        $this->update_comments();
    }
public function render()
    {
        $view = 'livewire.crm.caso.view';
        if($this->case['caso'] == 0){
            $view = 'livewire.crm.caso.invalid';
        }
        Log::info(sprintf('[%s] %s - view: %s',$this->ipaddr,__FUNCTION__,$view));
        Log::info(sprintf('[%s] %s - %d comments',$this->ipaddr,__FUNCTION__,count($this->cc)));
        return view($view)->layout('layouts.base');
    }
```

The issue is that I also have other components in the same "app" that are also bound to a route with a similar syntax, and they return data in the main livewire component and everything works just right. Is there any way to debug this any further?

**Context**
- Livewire version: 2.3.5
- Laravel version:  8.18.1
- Alpine version: 2.7.3
- Browser: Version 87.0.4280.88 (Official Build) (64-bit) / Firefox 82.0.3 (64-bit)

**Comment by austenc** [comment]
@joval-sv what are your `update_caso` and `update_comments` functions in the mount method doing? 

If you can provide a public repository where this error is occurring it will be much easier to help. Thanks!

**Comment by joval-sv** [comment]
These functions set/update data for various public properties used in 
sub-views to display relevant data by executing several queries to 
multiple databases to populate the properties, and I believe do not 
contribute to the problem. I have done this with other components 
without issues.

I would like to know where in the hydration process an error occurs that 
returns a NULL value to populate the wire:initial-data property.
